src/idl_gen_cpp.cpp

```diff
@@ -51,12 +54,28 @@ static std::string GeneratedFileName(const std::string &path,
 }
 
 namespace cpp {
+
+// Extension of IDLOptions for cpp-generator.
+struct IDLOptionsCpp : public IDLOptions {
```

Comment:
Our current IDLOptions already contains options for many languages, so no need to subclass I think

Reply:
The `cpp_std` is declared in IDLOptions, it can be set before call GenerateCPP if needed.
The `g_` prefixed options are specific for C++ generator only.
These options are experimental and only the cpp generator knows how to use them.
If these and future options will be fields of base IDLOptions then `flatc.cpp` should parse and set them, not `idl_gen_cpp.cpp`.

Reply:
ok